dont-worry-be-happy/dwbh-api
# Feat task 165 recover stats by group

Query to recover all the voting statistics,  given a group,  and between two dates.
https://tree.taiga.io/project/pabloruiz-kaleidos-patio/task/165?kanban-status=1813386

Validation tests
- [x] Run the graphql query with valid data to return results
- Check the paginated response for VotingStats is correct, changing the pageRequest info (page, max)
- Check the paginated response for VotingStats is ASC ordered (listing first the older statistics)
- Because the pagination have changed, check also the paginated response for Votes is correct changing its pageRequest info (page, max)
- [x] Repeat the query with wrong group, or out of range dates
- Check the returned VotingStats results are correct
- [x] Review the changes in pagination that may affect front 278
- [x] Move the task 165 to READY_FOR_TESTS

Query
```
query {
  getStatsByGroup(
    groupId: "d5847624-4b50-4d4a-abd3-ed9209a5448b"
    startDateTime: "2020-06-01T00:00:00Z"
    endDateTime: "2020-06-26T00:00:00Z"
    page: 0      #  <=====
    max: 10      #  <=====
  ) {
    page
    lastPage
    totalCount
    data {
      average
      movingAverage
      createdAtDateTime
      voting {
        id
        votes(page: 1, max: 2) {      #  <=====
          page
          lastPage
          totalCount
          data {
            score
          }
        }
      }
    }
  }
}
```

Response
```
{
  "data": {
    "getStatsByGroup": {
      "page": 0,
      "lastPage": 1,
      "totalCount": 17,
      "data": [
        {
          "average": 3.5,
          "movingAverage": null,
          "createdAtDateTime": "2020-06-26T12:06:48.900739+02:00",
          "voting": {
            "id": "6e65bbfc-c1ac-4555-ae87-8ef64dd492c3",
            "votes": {
              "page": 1,
              "lastPage": 2,
              "totalCount": 6,
              "data": [
                {
                  "score": 5
                },
                {
                  "score": 3
                }
              ]
            }
          }
        },
        {
          "average": 3.66,
          "movingAverage": null,
          "createdAtDateTime": "2020-06-26T12:06:48.900739+02:00",
          "voting": {
            "id": "5901b44a-0eb8-475f-96f4-50e7e2513761",
            "votes": {
              "page": 1,
              "lastPage": 2,
              ...
```
